### PR TITLE
Addition to PR#180 - new unit tests, jitter_timestamp changes

### PR DESCRIPTION
**Pull request by wetzelj**
Addition to PR#180 - new unit tests, jitter_timestamp changes

# Description

Related issues: #175 
Added unit tests to validate jittering of private tag fields as well as jitter of multiple fields identified by a values list.   Refactored jitter_timestamp to return the new jittered value and use replace_field to actually perform the modification of the field value. 

# Checklist

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] My code follows the style guidelines of this project

# Open questions

Questions that require more discussion or to be addressed in future development:

### CODE REVIEW

File: deid/dicom/actions.py [hunk 1]
```diff
@@ -23,47 +23,36 @@
...
-    dicom: the pydicom Dataset
     field: the field with the timestamp
     value: number of days to jitter by. Jitter bug!
 
     """
     if not isinstance(value, int):
         value = int(value)
 
-    original = dicom.get(field)
+    original = field.element.value
```
Comment (vsoch) [inline deid/dicom/actions.py:45]: Could this be none first? (field.element)

File: deid/dicom/actions.py [hunk 2]
```diff
@@ -23,47 +23,36 @@
...
+    new_value = original
 
     if original is not None:
 
         # Create default for new value
         new_value = None
-
-        # If we have a string, we need to create a DataElement
```
Comment (vsoch) [inline deid/dicom/actions.py:57]: hmm I think since we just pass field as is, it could still be a string?

File: deid/dicom/parser.py [hunk 3]
```diff
@@ -427,9 +427,8 @@ def _run_action(self, field, action, value=None):
             if value is not None:
 
                 # Jitter the field by the supplied value
-                jitter_timestamp(
-                    dicom=self.dicom, field=field.element.keyword, value=value
-                )
+                new_val = jitter_timestamp(field=field, value=value)
```
Comment (vsoch) [inline deid/dicom/parser.py:430]: Re my comment above: could field ever be a string here?

File: deid/dicom/actions.py (hunk 1 above)
Comment (wetzelj) [reply deid/dicom/actions.py:45]: I don't think so.  I checked the places that create DicomField objects, and they're checking to ensure that element is a DataElement prior to the creation of the DicomField.

File: deid/dicom/actions.py (hunk 2 above)
Comment (wetzelj) [reply deid/dicom/actions.py:57]: I don't believe so.  By the time we're performing this (via _run_action()), the string representation of the field has been evaluated to be a DicomField object by find_by_values() or expand_field_expression().   When parser.perform_action() calls _run_action(), we're iterating over a list of DicomFields.

File: deid/dicom/parser.py (hunk 3 above)
Comment (wetzelj) [reply deid/dicom/parser.py:430]: See above.

**Review by vsoch — APPROVED** [review]

**Pull request merged**